client/mirage/dhcp_ipv4.mli

```diff
@@ -16,6 +16,8 @@
 
 module Make (Network : Mirage_net.S) (E : Ethernet.S) (Arp : Arp.S) : sig
   include Tcpip.Ip.S with type ipaddr = Ipaddr.V4.t and type prefix = Ipaddr.V4.Prefix.t
-  val connect : Network.t -> E.t -> Arp.t -> t Lwt.t
+  val connect : ?cidr:Ipaddr.V4.Prefix.t -> ?gateway:Ipaddr.V4.t ->
+    ?options:Dhcp_wire.dhcp_option list -> ?requests:Dhcp_wire.option_code list ->
+    Network.t -> E.t -> Arp.t -> t Lwt.t
     (** Connect to an ipv4 device using information from a DHCP lease. *)
```

Comment:
The docstring should include some explanation of the behavior when you include the options to `connect` and when you don't.